src/configure.py

```diff
@@ -760,14 +769,50 @@ def load_config( config ):
             try:
                paths[temp[0]] = temp[1]
             except:
-               paths[temp[0]] = ''
+               paths[temp[0]] = ""
 
     return paths, compilers, flags, gpus
 
-def set_conditional_defaults( args ):
-    if args['machine'] is None:
-        args['machine'] = get_machine()
+def load_setting():
+    """
+    Return the default machine setting in the following order:
+    1. Read the local  setting located at `GAMER_LOCAL_SETTING`.
+    2. Read the global setting located at `GAMER_GLOBAL_SETTING`.
+    3. Fall back to `eureka_intel` for backward compatibility.
+
+    Format of the setting file:
+    1. Comment starts with `#`.
+    2. Separate the variable name and value with space. The line starts with the variable name.
+    3. Only the fisrt value of the line will be loaded.
+    4. Only the last variable of duplicated variables will be loaded.
+    """
+    sys_setting  = SystemSetting()
+    setting_file = None
 
+    if os.path.isfile( GAMER_GLOBAL_SETTING ): setting_file = GAMER_GLOBAL_SETTING
+    if os.path.isfile( GAMER_LOCAL_SETTING  ): setting_file = GAMER_LOCAL_SETTING
+
+    if setting_file is None:
+        LOGGER.info("System setting file not detected.")
+        return sys_setting
+    else:
+        LOGGER.info("Using %s as setting file."%(setting_file))
+
+    with open( setting_file, "r" ) as f:
+        lines = f.readlines()
+
+    for line in lines:
+        tokens = line.strip().split()
+        if len(tokens) == 0: continue      # empty line
+        if tokens[0][0] == "#": continue   # skip comment line
+        try:
+           sys_setting[tokens[0]] = tokens[1]
+        except:
+           sys_setting[tokens[0]] = None
```

Comment:
```suggestion
        if len(tokens) >= 2:
           sys_setting[tokens[0]] = tokens[1]
        else:
           sys_setting[tokens[0]] = None
```
I think in general we should avoid catch with non-explicitly exception.
Is there other case that will fail the suggestion?

Reply:
Agree, `if else` should be enough.
Can you also update these [lines](https://github.com/technic960183/gamer-fork-ymhsu/blob/b0573cee7917fe0c3f0d676f4f89aba1434360a8/src/configure.py#L760-L763) accordingly?

Reply:
Sure.